### ISSUE TITLE
ocamldoc replaces type variables with meaningless identifiers

### ISSUE THREAD

**Issue by vicuna** [ocaml/ocaml#7623]
**Original bug ID:** 7623
**Reporter:** @mjambon
**Status:** closed (set by @xavierleroy on 2017-10-03T12:01:37Z)
**Resolution:** unable to duplicate
**Priority:** normal
**Severity:** trivial
**Version:** 4.05.0
**Category:** ocamldoc
**Monitored by:** @Yakobowski

## Bug description

It would be good if ocamldoc preserved type variables so that we could have the following in the documentation:

<pre>
val get : ('key, 'value) Hashtbl.t -> 'key -> 'value option
</pre>

As it stands, the signature above gets converted into:

<pre>
val get : ('a, 'b) Hashtbl.t -> 'a -> 'b option
</pre>

What would it take to get this done?

**Comment by vicuna** [comment]
**Comment author:** @xclerc

I am unable to reproduce the problem (with a one-line mli file),
with versions 4.02.3 .. 4.05.0; is this declaration part of a
larger signature?

**Comment by vicuna** [comment]
**Comment author:** @xavierleroy

We need a repro case.  Thank you.

**Comment by vicuna** [comment]
**Comment author:** @mjambon

Indeed, it works for me. I used the command `ocamldoc -html toto.mli` with the signature above in the file `toto.mli` i.e. `val get : ('key, 'value) Hashtbl.t -> 'key -> 'value option`, with ocamldoc 4.04.1.

I reported this because @aantron (Anton Bachin) said it wasn't working, and I remembered from a long time ago that it used to not work.

All my apologies. I'll follow up on the original thread:

https://github.com/ocsigen/lwt/pull/469#issuecomment-327552611
